analysis/variables_cohorts.py

```diff
@@ -0,0 +1,584 @@
+            (last_matching_event_apc_before(
+                hypertension_icd10, index_date
+            ).exists_for_patient()) |
+            (last_matching_event_opa_before(
+                hypertension_icd10, index_date
+            ).exists_for_patient())
+        ),
+
+        ## Diabetes # to add hba1c?
+        cov_bin_diabetes=(
+            (last_matching_event_clinical_snomed_before(
+                diabetes_snomed_clinical, index_date
+            ).exists_for_patient()) |
+            (last_matching_med_dmd_before(
+                diabetes_drugs_dmd, index_date
+            ).exists_for_patient()) |
+            (last_matching_event_apc_before(
+                diabetes_icd10, index_date
+            ).exists_for_patient()) |
+            (last_matching_event_opa_before(
+                diabetes_icd10, index_date
+            ).exists_for_patient())
+        ),
+
+      ## Pneumonia
+        cov_bin_history_pneumonia_snomed= (
+            last_matching_event_clinical_snomed_before(
+            pneumonia_snomed, index_date
+            )
+        .exists_for_patient()
+        ),
+
+      ## Asthma
+        cov_bin_history_asthma_snomed= (
+            last_matching_event_clinical_snomed_before(
+            asthma_snomed, index_date
+            )
+        .exists_for_patient()
+        ),
+
+      ## Pulmonary Fibrosis
+        cov_bin_history_pulmonary_fibrosis_snomed= (
+            last_matching_event_clinical_snomed_before(
+            pulmonary_fibrosis_snomed, index_date
+            )
+        .exists_for_patient()
+        ),
+
+        ## Ischaemic stroke
+        cov_bin_stroke_isch=(
+            (last_matching_event_clinical_snomed_before(
+                stroke_isch_snomed_clinical, index_date
+            ).exists_for_patient()) |
+            (last_matching_event_apc_before(
+                stroke_isch_icd10, index_date
+            ).exists_for_patient()) |
+            (last_matching_event_opa_before(
+                stroke_isch_icd10, index_date
+            ).exists_for_patient())
+        ),
+
+        ## All stroke (included the all-stroke codelists)
+        cov_bin_all_stroke=(
+            (last_matching_event_clinical_ctv3_before(
+                stroke_codes, index_date
+            ).exists_for_patient()) |
+            (last_matching_event_clinical_snomed_before(
+                stroke_isch_snomed_clinical + stroke_sah_hs_snomed_clinical, index_date
+            ).exists_for_patient()) |           
+            (last_matching_event_apc_before(
+                stroke_isch_icd10 + stroke_sah_hs_icd10 + stroke_icd10, index_date
+            ).exists_for_patient()) |
+            (last_matching_event_opa_before(
+                stroke_isch_icd10 + stroke_sah_hs_icd10 + stroke_icd10, index_date
+            ).exists_for_patient())
+        ),
+# Others
+    ## History of Covid-19 Combined
+
+        tmp_sub_bin_priorcovid19_confirmed_sgss=tmp_sub_bin_priorcovid19_confirmed_sgss,
```

Comment:
Can the temporary variables be defined by the code that combines them or do they need to be outside?

Reply:
They need to be defined outside, so they can be included in the calculation here. Same as smoke variables. https://github.com/opensafely/post-covid-respiratory/pull/40#discussion_r1848038394